static/app/components/tabs/tabList.tsx

```diff
@@ -63,15 +66,29 @@ function useOverflowTabs({
     return () => observer.disconnect();
   }, [tabListRef, tabItemsRef]);
 
-  return overflowTabs;
+  const tabItemKeyToHiddenMap = tabItems.reduce(
+    (acc, next) => ({
+      ...acc,
+      [next.key]: next.hidden,
+    }),
+    {}
+  );
+
+  // Tabs that are hidden will be rendered with display: none so won't intersect,
+  // but we don't want to show them in the overflow menu
+  return overflowTabs.filter(tabKey => !tabItemKeyToHiddenMap[tabKey]);
 }
 
-interface TabListProps extends TabListStateProps<any>, AriaTabListProps<any> {
+interface TabListProps extends AriaTabListProps<TabListItemProps> {
```

Comment:
This change makes `items` the correct type, removed `TabListStateProps` because it is extended by `AriaTabListProps` already